### ISSUE TITLE
[core] GetAndroidApp not exposed

### ISSUE THREAD

**Issue by sogaiu** [raysan5/raylib#2364]
I'd like `GetAndroidApp` to be exposed for general use.

 - [X] I tested it on latest raylib version from master branch
 - [X] I checked there is no similar issue already reported
 - [X] My code has no errors or misuse of raylib

### Issue description

As noted [here in rcore.c](https://github.com/raysan5/raylib/blob/7fa75cb24e8fcc5983644276f9feed670e1f79c3/src/rcore.c#L713-L718):
```
// NOTE: Add this to header (if apps really need it)
struct android_app *GetAndroidApp(void)
{
    return CORE.Android.app;
}
#endif
```
`GetAndroidApp` is not exposed for general use.  With it exposed I am able to copy content from an `.apk` to a device's filesystem to aid in loading code (and other things) at runtime.  The code I'm loading is for the Janet programming language and for it to appropriately resolve import references, it's better to do this via a hierarchical filesystem rather than straight from the apk's assets.  It was very helpful to get at the relevant `android_app` via `GetAndroidApp`.

### Code Example

I made the following modification to `raylib.h` and it seemed to work:

```
diff --git a/src/raylib.h b/src/raylib.h
index 9f4e7caa..d54ffc06 100644
--- a/src/raylib.h
+++ b/src/raylib.h
@@ -1541,6 +1541,11 @@ RLAPI void SetAudioStreamPitch(AudioStream stream, float pitch);      // Set pit
 RLAPI void SetAudioStreamPan(AudioStream stream, float pan);          // Set pan for audio stream (0.5 is centered)
 RLAPI void SetAudioStreamBufferSizeDefault(int size);                 // Default size for new audio streams
 
+#if defined(PLATFORM_ANDROID)
+#include <android_native_app_glue.h>    // Required for: android_app struct and activity management
+RLAPI struct android_app *GetAndroidApp(void);
+#endif
+
 #if defined(__cplusplus)
 }
 #endif
```

Does this seem like the kind of change that would be good for Raylib?

**Comment by raysan5** [comment]
@sogaiu I'm afraid raylib does not expose any platform-dependant funtionality on it's API, feel free to implement it on your fork for your specific use case.

**Comment by sogaiu** [comment]
@raysan5 Ok, thanks!

**Comment by jn-jairo** [comment]
I just found out that using zig you just need to add this to your code to use the `GetAndroidApp()`.

```zig
const c = @cImport({
    @cInclude("android_native_app_glue.h");
});

extern fn GetAndroidApp() ?*c.struct_android_app;
```

I just tested adding it in the [zig-android-sdk raylib example](https://github.com/silbinarywolf/zig-android-sdk/tree/main/examples/raylib) and it works.

So, at least for zig you don't need to fork raylib to use the `GetAndroidApp()`.

I hope it may help someone.
